### PR TITLE
Comments branch

### PR DESCRIPTION
**Pull request by alexreigle**
I've built in the initial structure of doxygen comments. There are also updates to documents.

### CODE REVIEW

**Review by mbozada — COMMENTED** [review]
So I did a quick look through, but I mainly checked out the CommentsBranch and tried to build on my machine. "doxygen CraftDoc" went through, but it had a lot of extraneous information. As in, it generated documentation for several of the dependencies we're using. It made it kind of tricky to see what part of the project was ours.

On top of that, doing a regular make for the project didn't go through. However, in all things involving this crazy project, this may be done to incompetence or general error on my end.

Finally, I think it'd be good if you added more information to the initial pull request comment. I know that this is addressing documentation, but I wasn't sure what exactly you're trying to accomplish. Also, you've messed with some of your cloud code, but you haven't told me how you've changed things.

**Comment by alexreigle** [comment]
The documentation set up is furthering the coverage and usefulness of doxygen - I don't think it's too much of a problem that doxygen builds/documents everything. As in the email Prof. Buck sent us, we just need to denote the method to our madness in the README.md (i.e. we will either explicitly list every file and function there and how it meets the needed requirements or we will point to a doxygen page that does so). I kept the initial comment generic because there wasn't any doxygen code beforehand, so I sought to implement a sort of structure to guide the rest of our testing  requirements.

To get the project to build, you must do a fresh cmake command to include the test.cpp and test.h files.

The cloud files were changed in the course of testing and the changes will be reflected in the cloudBranch shortly, this is the objective of testing I'm pretty sure.

**Comment by alexreigle** [comment]
I have tentative plans to make the test protocol dependent on an input to craft (i.e. we call './build/craft(1)' or  './build/craft(test)'). How's that sound?

**Comment by mbozada** [comment]
That sounds like a potentially good approach to use. I'm nowhere near ready to test my stuff, and I haven't figured out a way to go about it. I'll leave it up to you.

Do you think it makes sense to hold off on this pull request until we're set on the final version of how we want to do some stuff? Basically, I think the only things we should be doing pull requests on are final or near-final products. However, this is a good sample for how documentation should proceed.

**Comment by alexreigle** [comment]
I'm ready and in the middle of testing, I think this is a good way to do so and also show that we have 100% test coverage - by listing all tasks completed and testing each one-by-one.

I only want to set the standard of testing, since you're leaving it up to me.

**Comment by alexreigle** [comment]
testing has now been implemented in a way that it only occurs if argument 'test' is passed with the executable call.

File: CraftDoc
Comment (mbozada) [inline CraftDoc:438]: This is the only thing I think we may want to change. My (very limited) understanding is that if we set this to no we'd only be generating doxygen documentation from properly formatted comments. On top of that, if we limit it to the header/source files we've made to isolate our work, the generated documentation would only include stuff that we've worked on.

**Review by mbozada — APPROVED** [review]
We've talked about this over discord. I think there may be some better options for documentation generation, but I'll focus on getting that done tomorrow.

For now, what's here looks good!

**Pull request merged**
